ml/src/main/scala/org/apache/spark/ml/h2o/algos/H2OGridSearch.scala

```diff
@@ -195,7 +195,7 @@ class H2OGridSearch(val gridSearchParams: Option[H2OGridSearchParams], override
     }
   }
 
-  private def sortGrid(grid: Grid[_]) = {
+  private def sortGrid(grid: Grid[ _ <: Model.Parameters]): Array[hex.Model[_, _ <: hex.Model.Parameters, _ <: hex.Model.Output]] = {
```

Comment:
Is the specification of the return type really needed?

Reply:
just tried - SW 2.1 on Scala 2.10 won't compile without this output type specified.

The automatically inferred types have slightly different type bounds leading to fail during compilation

Reply:
WDYT about using a type alias for the result type? IMHO, it would improve readability.

Reply:
Good point, especially for the long `hex.Model[_, _ <: hex.Model.Parameters, _ <: hex.Model.Output]`. Will do, thx!

Reply:
+1 , Hope there won't be any restrictions